sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -767,6 +767,13 @@ object SQLConf {
       .checkValue(_ > 0, "The initial number of partitions must be positive.")
       .createOptional
 
+  lazy val ALLOW_COLLATIONS_IN_MAP_KEYS =
+    buildConf("spark.sql.collation.allowInMapKeys")
+      .doc("Allow usage of collated strings inside of map's keys")
```

Comment:
any collated strings? I mean `str COLLATE UTF8_BINARY` is collated too. Should we say something like `Allow for non-UTF8_BINARY collated strings`